tests/Fixer/ClassNotation/FinalInternalClassFixerTest.php

```diff
@@ -191,6 +191,25 @@ class Bar {}
 ',
         ];
 
+        yield 'multiple classes, both with annotations, with one extending another class' => [
```

Comment:
this will work only if child class is reach-able by the Fixer.
By definition how Fixer works, this is not the case. Basically, while fixing Foo class, Fixer has no clue if Bar class exists and if it extends Foo or not.

Thus, very elaborated description of the rule.

I believe this PR tries to cover some cases and discover as much child-classes as possible, but from Fixer's architecture itself it's not possible.
Yet, regardless of the technical feasibility - rule is explicitly to force usage of `final` for classes: from docs: `No exception and no configuration are intentional`. This PR is breaking the purpose of the rule. It's like saying "whole codebase must follow given expectations, unlease they don't".